### PR TITLE
[#689] Remove major version for the database file name

### PR DESCRIPTION
**Pull request by ouziel-slama**
For review. Not tested yet.

### CODE REVIEW

File: counterpartylib/lib/blocks.py [hunk 1]
```diff
@@ -874,11 +874,21 @@ def follow(db):
         block_index = config.BLOCK_FIRST
     else:
         block_index = util.CURRENT_BLOCK_INDEX + 1
+        version_major, version_minor = util.get_user_version()
+        version_changed = False
```
Comment (adamkrellenstein) [inline counterpartylib/lib/blocks.py]: Do we really need to keep track of whether the version actually changed? Can't we just update `user_version` every time?

File: counterpartylib/lib/util.py [hunk 2]
```diff
@@ -600,4 +600,16 @@ def parse_id(match_id):
...
 
+
+def get_user_version():
+    user_version = cursor.execute('PRAGMA user_version').fetchall()[0]['user_version']
+    version_minor = user_version % 1000
+    version_major = int((user_version - version_minor) / 1000)
+    return version_major, version_minor
+
+def set_user_version(version_major, version_minor):
+    user_version = (int(version_major) * 1000) + int(version_minor)
```
Comment (adamkrellenstein) [inline counterpartylib/lib/util.py]: Won't `version_major` and `version_minor` always be `int`s already?

File: counterpartylib/lib/util.py [hunk 3]
```diff
@@ -600,4 +600,16 @@ def parse_id(match_id):
     assert match_id[64] == ID_SEPARATOR
     return match_id[:64], match_id[65:] # UTF-8 encoding means that the indices are doubled.
 
+
+def get_user_version():
+    user_version = cursor.execute('PRAGMA user_version').fetchall()[0]['user_version']
+    version_minor = user_version % 1000
+    version_major = int((user_version - version_minor) / 1000)
```
Comment (adamkrellenstein) [inline counterpartylib/lib/util.py]: Could we use `//` here instead?

**Comment by adamkrellenstein** [comment]
This line should be changed as well: https://github.com/CounterpartyXCP/counterpartyd/blob/master/test/util_test.py#L360

**Comment by adamkrellenstein** [comment]
Would it be too much work to add test coverage for this code now, before v9.49.4? Just so we're extra sure it works well. :)

File: counterpartylib/lib/blocks.py (hunk 1 above)
Comment (ouziel-slama) [inline counterpartylib/lib/blocks.py]: yeah you right, but in this case it becomes difficult to add a message in the log when there is a version change.

File: counterpartylib/lib/util.py (hunk 2 above)
Comment (ouziel-slama) [inline counterpartylib/lib/util.py]: yeah right! no need this `int()`

File: counterpartylib/lib/util.py (hunk 3 above)
Comment (ouziel-slama) [inline counterpartylib/lib/util.py]: yeah!

**Comment by ouziel-slama** [comment]
:-) sure! I will add tests for this.

File: counterpartylib/lib/blocks.py (hunk 1 above)
Comment (adamkrellenstein) [inline counterpartylib/lib/blocks.py]: Right. I think this is the best way to do it, then: 92bdc45

**Comment by ouziel-slama** [comment]
> This line should be changed as well: https://github.com/CounterpartyXCP/counterpartyd/blob/master/test/util_test.py#L360
already done here https://github.com/CounterpartyXCP/counterpartyd/blob/dbfile/counterpartylib/test/util_test.py#L382

**Comment by ouziel-slama** [comment]
@adamkrellenstein: I "slightly" modified the code. I created a function `check.database_version()`. What do you think to rename `check.version()` to `check.codebase_version()` ?

**Comment by adamkrellenstein** [comment]
`check.software_version()` is better, I think.

**Comment by ouziel-slama** [comment]
@adamkrellenstein: ok tests written and done. I think it's ok to merge.

**Comment by adamkrellenstein** [comment]
My computer's on the fritz. Feel free to merge.

**Comment by ouziel-slama** [comment]
Good luck!

**Pull request merged**